src/library_fs.js

```diff
@@ -913,6 +913,9 @@ FS.staticInit();` +
       if (!node.node_ops.setattr) {
         throw new FS.ErrnoError({{{ cDefine('EPERM') }}});
       }
+      if (node.node) {
+        node = node.node;
+      }
```

Comment:
What is this doing?  Should it happen up on line 911?

Reply:
I initially did this as permission checks would then be done on the FSStream object, and the FSNode object would only be use at the end once checks have been done. Though I overlooked the fact it may break correctness and is in fact only required for the ftruncate call, I'll group these checks with all the others and make a separate check for ftruncate